### DISCUSSION TITLE
How to stop puma-dev without uninstalling?

### DISCUSSION THREAD

**Discussion by alfuken** [puma/puma-dev#318] [Q&A] [answered]
I want to periodically run some other apps on port 80, but can not do so since puma-dev is occupying that port. `puma-dev -stop` only stops the apps, but does not stop the puma-dev itself. When I kill it, it automatically goes back up.

Despite being installed via brew, it is not possible to control `puma-dev` via `brew services` (which is a whole separate issue by itself). 

So the question is: how do I **temporarily** force puma-dev to stop and do NOT restart after being killed, until I explicitly want it back up again?

**Comment by nonrational** [comment, marked as answer]
`puma-dev -setup` configures itself via launchd as [`io.puma.dev`](https://github.com/puma/puma-dev/blob/b9d99890bfc9158b1cef4e31a34ab243bb7b3c0b/dev/setup_darwin.go#L131).

You should be able to suspend and resume the daemon using:

```shell
launchctl unload ~/Library/LaunchAgents/io.puma.dev.plist
launchctl load ~/Library/LaunchAgents/io.puma.dev.plist
```

Edit: Use load/unload instead of start/stop.

**Reply by alfuken** [reply]
Yes, well, not quite...

<img width="1728" alt="Screenshot 2022-08-26 at 19 57 52" src="https://user-images.githubusercontent.com/75377/186964285-c5459a04-2591-42c1-a61a-39b6e49a39c7.png">

**Reply by nonrational** [reply]
Sorry you're right. We don't want start/stop, we want load/unload.

```shell
launchctl unload ~/Library/LaunchAgents/io.puma.dev.plist
launchctl load ~/Library/LaunchAgents/io.puma.dev.plist
```

**Reply by alfuken** [reply]
Indeed, that worked! Thanks.

Would be a nice touch, though, if it would be possible to control it via brew services as well.
